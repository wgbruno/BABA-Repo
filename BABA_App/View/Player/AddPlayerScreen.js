import React, {useEffect, useState} from "react";
import {View, Text, StyleSheet, ScrollView, Alert} from 'react-native';
import CustomButton from "../../src/components/CustomButton";
import CustomInput from "../../src/components/CustomInput";
import { NavigationContainer } from 'react-native';
import { Player } from "../../Objects/PlayerCont";
import { getNewIDDB } from "../../DAOs/PlayerDao";
import { set } from "react-native-reanimated";
import { Account } from "../../Objects/AccountCont";


export default function AddPlayerScreen({navigation}){
    const [playerID, setID] = useState("");
    const [firstName, setFirst] = useState("");
    const [lastName, setLast] = useState("");
    const [number, setNumber] = useState(0);
    const [age, setAge] = useState(0);
    const [height, setHeight] = useState("");
    
    useEffect(() => {
        setID(getNewIDDB());
    })

    const onAddPlayer = () => {
        var player = new Player(playerID, firstName, lastName, parseInt(number), parseInt(age), height);
        if(!(player.createPlayer())){
            Alert.alert("Success!","You have created a new player!");
            var tmp = new Account(navigation.getParam("paramUserName", "N/A"));
            var account = tmp.getAccount();
            console.log(account);
            account.setPlayerID(playerID);
            if(navigation.getParam("paramAccountType", "N/A") == "Manager"){
              navigation.navigate("ManagerHome");
            }else{
              navigation.navigate("HomeScreen");
            }
        }
    }

    const onGoHome = () => {
      if(navigation.getParam("paramAccountType", "N/A") == "Manager"){
        navigation.navigate("ManagerHome");
      }else{
        navigation.navigate("HomeScreen");
      }
    }

    return (
        <ScrollView showsVerticalScrollIndicator={false}>
          <View style={styles.root}>
            <Text style={styles.title}>Create a player</Text>
    
            <CustomInput
              placeholder="First Name"
              value={firstName}
              setValue={setFirst}
            />
            <CustomInput 
              placeholder="Last Name" 
              value={lastName} 
              setValue={setLast} 
            />
            <CustomInput
              placeholder="Number"
              value={number}
              setValue={setNumber}
            />
            <CustomInput
              placeholder="Age"
              value={age}
              setValue={setAge}
            />
            <CustomInput
              placeholder="Height"
              value={height}
              setValue={setHeight}
            />
            <CustomButton 
              text="Add Player" 
              onPress={onAddPlayer} 
            />
            <CustomButton 
              text="Go Home" 
              onPress={onGoHome} 
            />

          </View>
        </ScrollView>
      );
};

const styles = StyleSheet.create({
    root: {
      alignItems: 'center',
      padding: 50,
    },
    title: {
      fontSize: 28,
      fontWeight: 'bold',
      color: '#051C60',
      margin: 10,
    },
    text: {
      color: 'gray',
      marginVertical: 30,
    },
    link: {
      color: '#FDB075',
    },
    container: {
      flex:1,
      alignItems: "center",
      justifyContent: "center",
    },
    checkboxContainer: {
      flexDirection: "row",
      marginBottom: 10,
    },
    checkbox: {
      alignSelf: "center",
    },
    label: {
      margin: 8,
    },
  });