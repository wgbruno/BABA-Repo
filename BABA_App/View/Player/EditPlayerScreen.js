import React, {useEffect, useState} from 'react';
import { NavigationContainer } from 'react-native';

export default function EditPlayerScreen({navigation}){
}