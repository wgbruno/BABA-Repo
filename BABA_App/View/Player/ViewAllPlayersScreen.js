import React, {useEffect, useState} from 'react';
import { Button, Text, TextInput, TouchableOpacity, View, StyleSheet, FlatList, Image, useWindowDimensions } from 'react-native';
import { NavigationContainer } from 'react-native';
import Logo from '../../assets/images/Logo_1.png'
import { getAllPlayers } from '../../DAOs/PlayerDao';

export default function ViewAllPlayersScreen({navigation}){
    const {height} = useWindowDimensions();
    const [players, setPlayers] = useState(getAllPlayers());
    const [empty, setEmpty] = useState(true);

    useEffect(() => {
        if(players != null){
            setEmpty(false);
        }
    })

    const noPlayers = () => {
        return(
            <View style={{justifyContent: 'center', alignItems: 'center', flex: 1}}>
                <Text style={{fontSize: 25, textAlign: 'center', color: 'darkblue'}}>
                    There are no players.
                </Text>
            </View>
        );
    }

    return(
        <View style={styles.root}>
        <Image
            source={Logo}
            style={[styles.logo, {height: height * 20}]}
            resizeMode="contain"
        />
        <Text style={{marginTop: 8, fontWeight: 'bold', color: 'darkblue', fontSize: 20}}>All Players</Text>
        {empty ? noPlayers():
            <FlatList
                data={players}
                keyExtractor={(item, index) => index.toString()}
                renderItem={({item, index}) => 
                    <View style={{flexDirection: 'row', justifyContent: 'space-evenly', margin: 10, flexGrow: 1}}>
                        <TouchableOpacity
                            onPress={() => {navigation.navigate('EditPlayer', {
                                paramFirstName: item.firstName,
                                paramLastName: item.lastName,
                                paramAge: item.age,
                                paramNumber: item.number,
                                paramHeight: item.height
                            });}}>  
                            <Text style={styles.list}>{item.firstName} {item.lastName}   Team: {item.teamName}</Text>
                            <Text style={styles.list}>Age: {item.age}      Height: {item.height}      Number: {item.number}</Text>
                        </TouchableOpacity>
                    </View>
                } 
            />}
        </View>
    );
};

const styles = StyleSheet.create({
    list: {
        marginTop: 8, 
        fontWeight: 'bold', 
        color: 'red', 
        fontSize: 20
    },
    root: {
      alignItems: 'center',
      padding: 10,
      flex: 1
    },
    logo: {
      width: '3000%',
      maxWidth: 200,
      maxHeight: 150,
    },
  });