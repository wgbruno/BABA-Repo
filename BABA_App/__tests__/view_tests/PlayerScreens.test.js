import {expect, jest, test} from '@jest/globals';
import renderer from 'react-test-renderer';
import AddPlayerScreen from '../../View/Player/AddPlayerScreen';
import EditPlayerScreen from '../../View/Player/EditPlayerScreen';
import ViewAllPlayersScreen from '../../View/Player/ViewAllPlayersScreen';

test("Add Player Screen", function () {
    const ap_tree = renderer.create(<AddPlayerScreen/>).toJSON();
    expect(ap_tree).toMatchSnapshot();
});
test("Edit Player Screen", function () {
    const ep_tree = renderer.create(<EditPlayerScreen/>).toJSON();
    expect(ep_tree).toMatchSnapshot();
});
test("View All Players Screen", function () {
    const vap_tree = renderer.create(<ViewAllPlayersScreen/>).toJSON();
    expect(vap_tree).toMatchSnapshot();
});