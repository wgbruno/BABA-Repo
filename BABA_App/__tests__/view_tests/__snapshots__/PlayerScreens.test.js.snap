// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Player Screen 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#051C60",
            "fontSize": 28,
            "fontWeight": "bold",
            "margin": 10,
          }
        }
      >
        Create a player
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="First Name"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Last Name"
          style={Object {}}
          value=""
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Number"
          style={Object {}}
          value={0}
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Age"
          style={Object {}}
          value={0}
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#e8e8e8",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginVertical": 5,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Height"
          style={Object {}}
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Add Player
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderRadius": 5,
              "marginVertical": 5,
              "padding": 10,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#DD4D44",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontWeight": "bold",
              },
              undefined,
              Object {},
            ]
          }
        >
          Go Home
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Edit Player Screen 1`] = `null`;

exports[`View All Players Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/images/Logo_1.png",
      }
    }
    style={
      Array [
        Object {
          "maxHeight": 150,
          "maxWidth": 200,
          "width": "3000%",
        },
        Object {
          "height": 26680,
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "darkblue",
        "fontSize": 20,
        "fontWeight": "bold",
        "marginTop": 8,
      }
    }
  >
    All Players
  </Text>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
